test/acc_SUITE.erl

```diff
@@ -133,38 +126,92 @@ produce_iq_meta_automatically(_C) ->
+    ?assertEqual(HostType, mongoose_acc:host_type(Acc3)),
+    ?assertEqual({FromJID, ToJID, El}, mongoose_acc:packet(Acc3)),
+    Ref = mongoose_acc:ref(Acc3),
+    NewServer = <<"test.", Server/binary>>,
+    NewHostType = <<"new ",HostType/binary>>,
+    NAcc1 = mongoose_acc:strip(#{lserver => NewServer,
+                                 host_type => NewHostType,
+                                 element => sample_stanza() }, Acc3),
     {XMLNS2, NAcc2} = mongoose_iq:xmlns(NAcc1),
     ?assertEqual(<<"urn:xmpp:blocking">>, XMLNS2),
     ?assertEqual(jid:from_binary(<<"a@localhost">>), mongoose_acc:to_jid(NAcc2)),
     ?assertEqual(Ref, mongoose_acc:ref(NAcc2)),
     ?assertEqual(997, mongoose_acc:get(ns, ppp, NAcc2)),
-    ?assertEqual([], mongoose_acc:get(ns2, NAcc2)).
-
+    ?assertEqual([], mongoose_acc:get(ns2, NAcc2)),
+    ?assertEqual(sample_stanza(), mongoose_acc:element(NAcc2)),
+    ?assertEqual(NewServer, mongoose_acc:lserver(NAcc2)),
+    ?assertEqual(NewHostType, mongoose_acc:host_type(NAcc2)),
+    NAcc3 = mongoose_acc:strip(Acc3),
+    {XMLNS3, NAcc4} = mongoose_iq:xmlns(NAcc3),
+    ?assertEqual(<<"urn:ietf:params:xml:ns:xmpp-session">>, XMLNS3),
+    ?assertEqual(<<"set">>, mongoose_acc:stanza_type(NAcc3)),
+    ?assertEqual(Server, mongoose_acc:lserver(NAcc4)),
+    ?assertEqual(HostType, mongoose_acc:host_type(NAcc4)),
+    ?assertEqual({FromJID, ToJID, El}, mongoose_acc:packet(NAcc4)),
+    ?assertEqual(Ref, mongoose_acc:ref(NAcc4)),
+    ?assertEqual(997, mongoose_acc:get(ns, ppp, NAcc4)),
+    ?assertEqual([], mongoose_acc:get(ns2, NAcc4)).
+
+tries_to_get_host_type(_Config) ->
+    meck:new(mongoose_domain_api),
+    meck:expect(mongoose_domain_api, get_host_type,
+                fun
+                    (<<"host1">>) -> {ok, <<"host 1">>};
+                    (<<"host2">>) -> {ok, <<"host 2">>};
+                    (_) -> {error, not_found}
+                end),
+    AccParams=#{location => ?LOCATION},
+    StripParams = #{element => sample_stanza()},
+    %% new with <<"host 1">> host type
+    Acc1 = mongoose_acc:new(AccParams#{lserver => <<"host1">>}),
+    ?assertEqual(<<"host 1">>, mongoose_acc:host_type(Acc1)),
+    ?assertEqual(1, meck:num_calls(mongoose_domain_api, get_host_type, [<<"host1">>])),
+    %% new with 'undefined' host type
+    Acc2 = mongoose_acc:new(AccParams#{lserver => <<"unknown.host1">>}),
+    ?assertEqual(undefined, mongoose_acc:host_type(Acc2)),
+    ?assertEqual(1, meck:num_calls(mongoose_domain_api, get_host_type, [<<"unknown.host1">>])),
+    %% host type changes from <<"host 1">> to 'undefined'
+    Acc3 = mongoose_acc:strip(StripParams#{lserver => <<"unknown.host1">>},Acc1),
+    ?assertEqual(undefined, mongoose_acc:host_type(Acc3)),
+    ?assertEqual(2, meck:num_calls(mongoose_domain_api, get_host_type, [<<"unknown.host1">>])),
+    %% host type changes from <<"host 1">> to <<host 2>>
+    Acc6 = mongoose_acc:strip(StripParams#{lserver => <<"host2">>},Acc1),
```

Comment:
The ordering does not help either: Acc1, Acc2, Acc3, Acc6, Acc4, Acc5

Reply:
yeah, i've decided to move Acc6 test but forgot to change the ids.